sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -414,6 +409,23 @@ object GeneratorNestedColumnAliasing {
       None
   }
 
+  /**
+   * Replace the refernece attribute of extractor expression with generator input.
+   */
+  private def replaceGenerator(generator: ExplodeBase, expr: Expression): Expression = {
+    expr.transformDown {
+      case _: Attribute =>
+        generator.child
```

Comment:
Since we switch to `transformDown` from `transformUp`, this could be revisited and transformed? Is it safe from recursion?

Reply:
Oh, this is good point. For `GetStructField`, because we simply replace with another `ExtractValue`, I think we are fine. For `Attribute` case, if `generator.child` is not an attribute but, for example, another `ExtractValue`, we might wrongly recursively transform down.

We should add a guard here. I will update it too.